splinterd/tests/admin/circuit_abandon.rs

```diff
@@ -483,3 +483,336 @@ pub fn test_3_party_circuit_abandon() {
+        &circuit_id,
+        node_a.node_id(),
+        &*node_a.admin_signer().clone_box(),
+    );
+    assert!(node_a
+        .admin_service_client()
+        .submit_admin_payload(abandon_payload)
+        .is_ok());
+
+    // Restart the second node in the network
+    network = network.start(1).expect("Unable to start second node");
+    node_a = network.node(0).expect("Unable to get first node");
+    node_b = network.node(1).expect("Unable to get second node");
+    node_c = network.node(2).expect("Unable to get third node");
+
+    assert_eq!(
+        node_a
+            .admin_service_client()
+            .list_circuits(Some("status=abandoned"))
+            .expect("Unable to list circuits")
+            .data
+            .len(),
+        1
+    );
+    assert_eq!(
+        node_b
+            .admin_service_client()
+            .list_circuits(None)
+            .expect("Unable to list circuits")
+            .data
+            .len(),
+        2
+    );
+    assert_eq!(
+        node_c
+            .admin_service_client()
+            .list_circuits(None)
+            .expect("Unable to list_circuits")
+            .data
+            .len(),
+        2
+    );
+
+    // Stop the third node in the network
+    network = network.stop(2).expect("Unable to stop third node");
+    node_b = network.node(1).expect("Unable to get second node");
+
+    // Create the abandon request to be sent from the second node
+    let abandon_payload = make_circuit_abandon_payload(
+        &circuit_id,
+        node_b.node_id(),
+        &*node_b.admin_signer().clone_box(),
+    );
+    assert!(node_b
+        .admin_service_client()
+        .submit_admin_payload(abandon_payload)
+        .is_ok());
+
+    // Restart the third node in the network
+    network = network.start(2).expect("Unable to start third node");
+    node_a = network.node(0).expect("Unable to get first node");
+    node_b = network.node(1).expect("Unable to get second node");
+    node_c = network.node(2).expect("Unable to get third node");
+
+    assert_eq!(
+        node_a
+            .admin_service_client()
+            .list_circuits(Some("status=abandoned"))
+            .expect("Unable to list circuits")
+            .data
+            .len(),
+        1
+    );
+    assert_eq!(
+        node_b
+            .admin_service_client()
+            .list_circuits(Some("status=abandoned"))
+            .expect("Unable to list circuits")
+            .data
+            .len(),
+        1
+    );
+    assert_eq!(
+        node_c
+            .admin_service_client()
+            .list_circuits(None)
+            .expect("Unable to list_circuits")
+            .data
+            .len(),
+        2
+    );
+
+    // Stop the first node in the network
+    network = network.stop(0).expect("Unable to stop first node");
+    node_c = network.node(2).expect("Unable to get third node");
+
+    // Create the abandon request to be sent from the second node
```

Comment:
looks like this abandon request is being sent from the third node

Reply:
Fixed this comment